text/0507-subnets.md

```diff
@@ -0,0 +1,402 @@
+});
+```
+
+If you add more than one subnet to the VPC, the framework validates that there
+is no intersection between their address blocks. In addition, if all VPC IP
+address blocks (both primary and secondary) are provided as CIDR strings, the
+framework validates that each address block of all subnets is within one of the
+address blocks of the VPC.
+
+If you have added a secondary IPv6 block to your VPC, you can then add
+subnets with IPv6 ranges as well:
+
+```ts
+const subnet = vpc.addSubnet('subnet', {
+  cidrBlock: '2001:db8:1234:1a00::/60',
+  availabilityZone: 'us-west-2a'
+});
+```
+
+### Routing
+
+By default, `addSubnet()` creates isolated subnets, that only route traffic
+to other hosts inside the VPC. To define different routing policies for a
+subnet, provide a route table when creating it. For example, to create a
+public subnet:
+
+```ts
+const publicRouteTable = vpc.addRouteTable('routeTable', {
+  routes: [
+    // With no parameters, the destination is the entire VPC CIDR
+    Route.local(),
+
+    // By adding this route, all subnets that use this table become public
+    Route.to({
+      destination: '0.0.0.0/0',
+      target: Routers.INTERNET_GATEWAY,
+    }),
+  ],
+});
+
+const subnet = vpc.addSubnet('publicSubnet', {
+  cidrBlock: '10.2.0.0/20',
+  availabilityZone: 'us-west-2a',
+  routeTable: publicRouteTable,
+  mapPublicIpOnLaunch: false, // default: true for public subnets
+});
+
+// The following is true
+vpc.publicSubnets.includes(subnet);
+
+// As is
+vpc.selectSubnets({subnetType: ec2.SubnetType.PUBLIC}).includes(subnet);
+```
+
+To create a route table that sends DynamoDB and S3 traffic through VPC
+endpoints:
+
+```ts
+const routeTable = new RouteTable(this, 'routeTable', {
+  routes: [
+    Route.DYNAMODB_ENDPOINT,
+    Route.S3_ENDPOINT
+  ],
+});
+```
+
+You can use a public NAT gateway to enable instances in a private subnet to
+send outbound traffic to the internet, while preventing the internet from
+establishing connections to the instances:
+
+```ts
+const elasticIp = new ElasticIp({
+  domain: Domain.VPC
+
+  // Other properties, such as networkBorderGroup and publicIpv4Pool, 
+  // are also available. Omitted here for brevity.
+});
+
+const natGateway = publicSubnet.addNatGateway('natgw', {elasticIp});
+
+const routeTable = vpc.addRouteTable('routeTable', {
+  routes: [
+    Route.to({
+      destination: '0.0.0.0/0',
+
+      // targets must implement the IRouter interface
+      target: natGateway,
+    }),
+  ],
+});
+
+const privateSubnet = vpc.addSubnet('privateSubnet', {
+  cidrBlock: '10.2.0.0/20',
+  availabilityZone: 'us-west-2a',
+  routeTable,
+});
+```
+
+You can also create the same kind of routing pattern, but with a NAT instance
+instead:
+
+```ts
+const natInstance = publicSubnet.addNatInstance('natinst', {
+  instanceType: ec2.InstanceType.of(ec2.InstanceClass.BURSTABLE2, ec2.InstanceSize.MICRO),
+  machineImage: ec2.MachineImage.latestAmazonLinux({
```

Comment:
Wait don't you need a special AMI to serve as NAT instance?

Reply:
Yes, you're right. I picked a random example, and guess what: it doesn't make sense 🤦